runtime/foucoco/src/lib.rs

```diff
@@ -1293,18 +1293,30 @@ where
 				let mut env = env.buf_in_buf_out();
 				let price_feed_request: ([u8; 32], [u8; 32]) = env.read_as()?;
 
-				let blockchain = price_feed_request.0;
-				let symbol = price_feed_request.1;
-				let price_feed = <dia_oracle::Pallet<T> as DiaOracle>::get_coin_info(
-					blockchain.to_vec(),
-					symbol.to_vec(),
-				);
+				let blockchain = trim_trailing_zeros(&price_feed_request.0).to_vec();
+				let symbol = trim_trailing_zeros(&price_feed_request.1).to_vec();
 
-				warn!("price_feed_request : {:#?}", price_feed_request);
-				warn!("price_feed : {:#?}", price_feed);
+				let result =
+					<dia_oracle::Pallet<T> as DiaOracle>::get_coin_info(blockchain, symbol);
 
-				env.write(&price_feed.encode(), false, None).map_err(|_| {
-					DispatchError::Other("ChainExtension failed to call price feed")
+				warn!("price_feed_request : {:#?}", price_feed_request);
+				warn!("price_feed : {:#?}", result);
+
+				let result = match result {
+					Ok(ok) =>
```

Comment:
renamed 👍 we do need it because dia's CoinInfo doesn't derive scale Encode/Decode nor does it derive MaxEncodedLength. These are needed on any types to be communicated between a wasm contract and the chain.